source/microbit/help.c

```diff
@@ -68,61 +69,110 @@ STATIC const mp_doc_t help_table_types[] = {
+    {&microbit_image_width_obj, "Return the width of the image in pixels.\n"},
+    {&microbit_image_height_obj, "Return the height of the image in pixels.\n"},
+    {&microbit_image_get_pixel_obj, "Use get_pixel(x, y) to return the image's brightness at LED pixel (x,y).\nBrightness can be from 0 (LED is off) to 9 (maximum LED brightness).\n"},
+    {&microbit_image_set_pixel_obj, "Use set_pixel(x, y, b) to set the LED pixel (x,y) in the image to brightness\n'b' which can be set between 0 (off) to 9 (full brightness).\n"},
+    {&microbit_image_shift_left_obj, "Use shift_left(i) to make a copy of the image but moved 'i' pixels to the left.\n"},
+    {&microbit_image_shift_right_obj, "Use shift_right(i) to make a copy of the image but moved 'i' pixels to\nthe right.\n"},
+    {&microbit_image_shift_up_obj, "Use shift_up(i) to make a copy of the image but moved 'i' pixels up.\n"},
+    {&microbit_image_shift_down_obj, "Use shift_down(i) to make a copy of the image but moved 'i' pixels down.\n"},
+    {&microbit_image_copy_obj, "Use copy() to make a new exact copy of the image.\n"},
+    {&microbit_image_crop_obj, "Use crop(x1, y1, x2, y2) to make a cut-out copy of the image where coordinate\n(x1,y1) is the top left corner of the cut-out area and coordinate (x2,y2) is the\nbottom right corner.\n"},
+    {&microbit_image_invert_obj, "Use invert() to make a negative copy of the image. Where a pixel was bright or\non in the original, it is dim or off in the negative copy.\n"},
+    // uart
+    {&microbit_uart_obj, "Communicate with a serial device connected to micro:bit's I/O pins.\n"},
+    {&microbit_uart_init_obj, "Use init() to set up communication. Use pins 0 (TX) and 1 (RX) with a baud\nrate of 9600.\nOverride the defaults for 'baudrate', 'parity' and 'pins'.\n"},
+    {&microbit_uart_any_obj, "If there are incoming characters waiting to be read, any() will return True.\nOtherwise, returns False.\n"},
+    {&mp_stream_read_obj, "Use read() to read characters.\nUse read(n) to read, at most, 'n' bytes of data.\n"},
+    {&mp_stream_readall_obj, "Use readall() to read as much data as possible.\n"},
+    {&mp_stream_unbuffered_readline_obj, "Use readline() to read a line that ends with a newline character.\n"},
+    {&mp_stream_readinto_obj, "Use readinto(buf) to read bytes into the buffer 'buf'.\nUse readinto(buff, n) to read, at most, 'n' number of bytes into 'buf'.\n"},
+    {&mp_stream_write_obj, "Use write(buf) to write the bytes in buffer 'buf' to the connected device.\n"},
+    // Music module
+    {&music_module, "Plug in a speaker with crocodile clips and make micro:bit go bleep and bloop.\n"},
+    {&microbit_music_set_tempo_obj, "Use set_tempo(n, bpm) to make a beat 'n' ticks long and play at 'bpm' beats\nper minute.\n"},
```

Comment:
So... the API docs explain it (viz. http://microbit-micropython.readthedocs.org/en/latest/music.html). But in summary, each beat can be subdivided into `number` equal units. So, if my piece was in 4/4 and the fastest note I needed to play was a 16th note (or semiquaver as we say over here), then `number=4` because a crotchet can be subdivided into 4 semiquavers. The BPM tells me how many time a beat will be played in a minute (say 120 for an Allegro moderato). That means, 120*4=the number of 16th notes playable in a minute, or 480.

Does this make sense? Music is awesome, especially when it's so wonderfully mathematical like this. Such a great way to intuitively "grok" fractions. ;-)